pkg/agent/agent.go

```diff
@@ -189,20 +191,37 @@ func startAPIServiceCache() {
 		log.Errorf("could not start the New APIs cache update job: %v", err.Error())
 		return
 	}
+
 	log.Tracef("registered API cache update job: %s", id)
+
 	if !agent.useGrpc {
 		// Start the full update after the first interval
-		go func() {
-			time.Sleep(time.Hour)
-			allDiscoveryCacheJob := newDiscoveryCache(true)
-			id, err := jobs.RegisterIntervalJobWithName(allDiscoveryCacheJob, time.Hour, "All APIs Cache")
-			if err != nil {
-				log.Errorf("could not start the All APIs cache update job: %v", err.Error())
-				return
-			}
-			log.Tracef("registered API cache update all job: %s", id)
-		}()
+		go startDiscoveryCache()
+		return
 	}
+
+	host := agent.cfg.GetURL()
+	tenantID := agent.cfg.GetTenantID()
+	insecure := agent.cfg.GetTLSConfig().IsInsecureSkipVerify()
+	c := stream.NewClient(
+		host,
+		tenantID,
+		"/management/v1alpha1/watchtopics/mock-watch-topic",
+		insecure,
+		agent.cfg.GetTLSConfig(),
+		agent.tokenRequester,
+		stream.NewAPISvcHandler(agent.apiMap),
+		stream.NewInstanceHandler(cache.New()),
+		stream.NewCategoryHandler(agent.categoryMap),
+		// TODO: agents should be able to pass in their own handlers.
+	)
+
+	go func() {
+		err := c.Start()
+		if err != nil {
+			log.Errorf("failed to start the grpc client: %s", err)
+		}
+	}()
```

Comment:
Can we pull this out into a separate function like the discovery cache function?

Reply:
Yes